### DISCUSSION TITLE
clarification needed: Deployment of Qt backend application

### DISCUSSION THREAD

**Discussion by Kuraga13** [slint-ui/slint#3187] [General]
I have just compiled a minimal Rust example with the Slint library. Connecting Qt was quite tricky (mingw does not contain QtCore.lib, so I had to choose msvc), but thanks to my experience with Qt, I managed to figure it out pretty quickly.

Regarding deployment - in Qt, this is perhaps the most confusing and unpleasant part. For example, in native qt, I had to specify the necessary dlls in the .pro projects, in the install files, and add install arguments to the makefile.

How will this be handled in Slint? For any user without Qt experience, deploying the exe on any machine will be a nightmare.

**Comment by tronical** [comment]
That's a good topic. We don't have a good deployment story/tooling for the desktop when Qt is a dependency. In principle it's a little easier because we need only three DLLs (core, gui, windows platform), but the devil is - as you surely have experienced - in the detail (and directory structure).

A starting point would perhaps be a deployment chapter in the Rust and C++ docs (with some things in common and others not).

Deployment without Qt should be a little easier. With Rust you get a .exe file and that's it. With C++ at the moment we're still using dynamic linkage so you also need libslint_cpp.dll.

What's your experience? Any particular advice?

**Comment by Kuraga13** [comment]
Overall, my experience is very limited, but the main point is that I have some story deploying Qt. 

Yes you are right, for starting point, I think it would be helpful to provide a simple package and directory structure description in the documentation, as many new users may not be familiar with what's "under the hood" and may not have experience with Qt.

I'm not sure, but in the future it might be useful to include a DLL selector directly in the build script so that everything necessary is in the target folder. How it seemed to me cxx_qt is "under the hood," it may be worth using installs in cmake.txt and install arguments in cmake, but you are much more competent in this area than I am.
